### DISCUSSION TITLE
How do you transcribe a long audio file?

### DISCUSSION THREAD

**Discussion by nickponline** [openai/whisper#842] [Q&A]
I have a 30 min mp3 I would line to transcribe with whisper is there a command line that will do this, or chunk it up automatically?

**Comment by abhinavkulkarni** [comment]
@nickponline: Chunking is done automatically.

**Reply by nickponline** [reply]
I see, so you cant get the transcription on the fly during processing. You have to wait for the whole file to be processed?

**Reply by dgalli1** [reply]
No you can just grab it directly from stdout.

**Reply by masafumimori** [reply]
I get this error when i try to transcribe a large audio (like one hour) file...
```python
413 {'error': {'message': 'Maximum content size limit (26214400) exceeded (30683552 bytes read)'
```
Do you know how to manage such a large file?

Code I used：https://colab.research.google.com/drive/15H4GkjE5GfwrkS7w3HqqkBcvZvwuc8sD?usp=sharing#scrollTo=WF8l3NaIMkwe

**Reply by jongwook** [reply]
@nickponline We're thinking of supporting a callback or making a generator version of `transcribe()`  (some discussions in #1025). 

@masafumimori The OP was about using this Python package and model locally, and the 25MiB limit is a temporary restriction on the maximum file size when using the Whisper API. While we'd like to increase the limit in the future (or better, make an async API), you could split the MP3 in the meantime, or convert to a lower-bitrate MP3 or M4A to make it under 25MiB and submit it to the API for now.

**Reply by masafumimori** [reply]
@jongwook 
I see! I tried that and it worked! Thank you :)

**Reply by niuta24** [reply]
@masafumimori can you share your code, please? did you do some code to automatically divide audio at 25Mb parts and give them to Whisper?

**Reply by masafumimori** [reply]
@niuta24 
I did something like [this](https://gist.github.com/masafumimori/16bb3509cc0d514d57a91ebb7384f256).

In my case, it was enough to split into two.
Also, be aware of doing this may split at the middle of sentence being spoken.
Hope this helps!

**Reply by SpyCoder77Alt** [reply]
is there a way to merge the two transcriptions into one?

**Comment by phineas-pta** [comment]
you can use ffmpeg to split the audio, for e.g. making 1h segment `ffmpeg -hide_banner -i <YOUR AUDIO FILE> -c copy -map 0 -segment_time 1:0:0 -f segment <OUTPUT DIR>/segment_%03d`

and then you can loop over the directory with whisper

**Reply by omenking** [reply]
I would think that this could lead inaccuraries if you are spliting the middle of an utterance, and so I guess you have to have overlap between chunks to get best results?

**Reply by SpyCoder77Alt** [reply]
yeah
